src/biome/text/_model.py

```diff
@@ -377,8 +378,29 @@ def _setup(self):
         vocab = self._model.vocab
         vocab.save_to_files(os.path.join(self._serialization_dir, "vocabulary"))
 
-        self._iterator = DataIterator.from_params(self._params.pop("iterator"))
-        self._iterator.index_with(vocab)
+        #self._iterator = DataIterator.from_params(self._params.pop("iterator"))
+        #self._iterator.index_with(vocab)
+        iterator_config = self._params.pop("iterator")
+        
+        for dataset in self._all_datasets.values():
+            dataset.index_with(vocab)
+        
+        self._dataloader = DataLoader(
+            dataset=self._all_datasets["train"],
+            batch_size=iterator_config["batch_size"]
```

Comment:
@frascuchon @dcfidalgo 
I included now the batch_size. 

PyTorch DataLoaders do not allow custom batch_samplers (for using previous iterator params for bucketing) for IterableDatasets, this is a known issue  https://github.com/allenai/allennlp/issues/3966 but frankly I think we can live without bucketing for now for the type of problems we are tackling. We can maybe discuss this later today.